gitpod-io/website
# Fix Svelte/Vite HMR config

## Description

There's an issue preventing us from working with this repository out of the box using VS Code Desktop: the website enters into an endless refreshing loop. ([Internal discussion](https://gitpod.slack.com/archives/C020VCB0U5A/p1660145249168809))

This PR allows it to correctly work on VS Code Browser and Desktop IDEs by forcing Vite HMR websocket to connect to through Gitpod Port URL:

| Running without port forwarding (VS Code Browser) | Running with port forwarding (Desktop IDEs) |
| --- | --- |
| <img width="672" alt="image" src="https://user-images.githubusercontent.com/418083/184135861-80fbb6ec-7690-466d-b754-a6844a29a4e5.png"> | <img width="682" alt="image" src="https://user-images.githubusercontent.com/418083/184135847-6989839c-9336-4f10-b9c1-c1dc2f94a62e.png"> |

## How to test

1. Go to [Gitpod.io Preferences](https://gitpod.io/preferences), select VS Code Browser as Editor
2. Start a workspace using this URL: https://gitpod.io/#https://github.com/gitpod-io/website/pull/2562
3. Connect to the workspace using VS Code Desktop
4. Run this command on terminal: `gp preview --external $(gp url 3000)`, confirm to open the URL in the browser, and check if the website is not in a refresh-loop.
5. In another browser tab, visit http://localhost:3000/ and check if the website is not in a refresh-loop.
6. With these two tabs opened, replace "Always" with "Eternally" on `src/lib/components/index/hero.svelte` and confirm if the website is updated in both tabs.
    | | |
    | --- | --- |
    | <img width="311" alt="image" src="https://user-images.githubusercontent.com/418083/184138982-9e294997-6a2b-4c81-94a1-f18a9bbfef72.png"> | <img width="378" alt="image" src="https://user-images.githubusercontent.com/418083/184140265-6a30f4ab-8c14-41e2-854f-44f977d0b0de.png"> |





<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2562"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
